autogpt_platform/backend/backend/blocks/google_maps.py

```diff
@@ -0,0 +1,125 @@
+
+
+class GoogleMapsSearchBlock(Block):
+    class Input(BlockSchema):
+        api_key: BlockSecret = SecretField(
+            key="google_maps_api_key",
+            description="Google Maps API Key",
+        )
+        query: str = SchemaField(
+            description="Search query for local businesses",
+            placeholder="e.g., 'restaurants in New York'",
+        )
+        radius: int = SchemaField(
+            description="Search radius in meters (max 50000)",
+            default=5000,
+            ge=1,
+            le=50000,
+        )
+        max_results: int = SchemaField(
+            description="Maximum number of results to return",
+            default=20,
+            ge=1,
+            le=60,
+        )
+
+    class Output(BlockSchema):
+        businesses: List[dict] = SchemaField(description="List of businesses found")
+        error: str = SchemaField(description="Error message if the search failed")
+
+    def __init__(self):
+        super().__init__(
+            id="f47ac10b-58cc-4372-a567-0e02b2c3d479",
+            description="This block searches for local businesses using Google Maps API.",
+            categories={BlockCategory.SEARCH},
+            input_schema=GoogleMapsSearchBlock.Input,
+            output_schema=GoogleMapsSearchBlock.Output,
+            test_input={
+                "api_key": "your_test_api_key",
+                "query": "restaurants in new york",
+                "radius": 5000,
+                "max_results": 5,
+            },
+            test_output=[
+                (
+                    "businesses",
+                    [
+                        {
+                            "name": "Test Restaurant",
+                            "address": "123 Test St, New York, NY 10001",
+                            "phone": "+1 (555) 123-4567",
+                            "rating": 4.5,
+                            "reviews": 100,
+                            "website": "https://testrestaurant.com",
+                        }
+                    ],
+                ),
+            ],
+            test_mock={
+                "search_places": lambda *args, **kwargs: [
+                    {
+                        "name": "Test Restaurant",
+                        "address": "123 Test St, New York, NY 10001",
+                        "phone": "+1 (555) 123-4567",
+                        "rating": 4.5,
+                        "reviews": 100,
+                        "website": "https://testrestaurant.com",
+                    }
+                ]
+            },
+        )
+
+    def run(self, input_data: Input) -> BlockOutput:
+        try:
+            businesses = self.search_places(
+                input_data.api_key.get_secret_value(),
+                input_data.query,
+                input_data.radius,
+                input_data.max_results,
+            )
+            yield "businesses", businesses
```

Comment:
why is it businesses ? Isn't it supposed to be a generic places ?

Reply:
because its searching google maps for businesses in the place to return the info for it, I should probs rename the block

Reply:
Ohh is it actually business specific?

Or can I use this block to search landmarks or nature

Reply:
it is business specific, but i do want to add another block for searching for landmarks and so on

Reply:
well its a bit weird, i typed in "new york" and i got back

```
[{"name":"New York","address":"New York, NY, USA","phone":"","rating":0,"reviews":0,"website":"http://www.nyc.gov/"}]
```

Reply:
Let's just call it place then? 😄
Also we might want to make it singular and yield dicts instead of producing list in a single go.
It's simpler that we don't need another iterator block

Reply:
So change it from businesses to "place" or "places"(this feels better)

And yeah ok ill update it and make those changes

Reply:
Yeah places work, as long as we are not returning a list.